compiler/rustc_lint/src/context.rs

```diff
@@ -865,9 +865,15 @@ pub trait LintContext: Sized {
 
                     if let Some(positional_arg_to_replace) = position_sp_to_replace {
                         let name = if is_formatting_arg { named_arg_name + "$" } else { named_arg_name };
-
+                        let span_to_replace = if let Ok(positional_arg_content) = self.sess().source_map().span_to_snippet(positional_arg_to_replace) {
+                            if positional_arg_content.contains(":?") {
```

Comment:
We also need to take into account something like `:#?`.

https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=1f095b3602a2b95d8637b47aa357fc5b

Reply:
Ah! right! maybe we should check if a string start `:` and end `?` sounds like a regex job?

Reply:
I think just checking if `positional_arg_content` starts with `:` can solve this problem. Ideally, we should improve `PositionalNamedArg::get_positional_arg_spans` so that it returns appropriate `position_sp_to_replace` by improving `has_formatting` field of `PositionalNamedArg`, but I think there is no need to do this much.

https://github.com/rust-lang/rust/blob/a8a33cf27166d3eabaffc58ed3799e054af3b0c6/compiler/rustc_builtin_macros/src/format.rs#L68-L101

Reply:
I think that we can open an issue and move this to another task? I think your solution is the better one to improve the code, and remove some if else.

Reply:
Tracking here https://github.com/rust-lang/rust/issues/100891 I will give the love that it needed in a new PR

Reply:
Thank you!